app/connectors/NpsConnector.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.{ApplicationConfig, WSHttp}
+import play.api.Mode.Mode
+import play.api.http.Status.{BAD_GATEWAY, GATEWAY_TIMEOUT, INTERNAL_SERVER_ERROR, OK}
+import play.api.libs.json.JsValue
+import play.api.{Configuration, Logger, Play}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpGet, HttpReads, _}
+import uk.gov.hmrc.play.config.ServicesConfig
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object NpsConnector extends NpsConnector with ServicesConfig {
+
+  override val serviceUrl = ApplicationConfig.npsServiceUrl
+  override def http = WSHttp
+
+}
+
+trait NpsConnector {
+
+  protected def mode: Mode = Play.current.mode
+  protected def runModeConfiguration: Configuration = Play.current.configuration
```

Comment:
These are defined for `ServicesConfig` but the trait doesnt know there is a relationship for these values. I think they're better placed in the object, unless you make the trait implement ServiceConfig (though, the trait probably shouldnt care)

Reply:
done